mozdownload/scraper.py

```diff
@@ -274,8 +279,14 @@ def total_seconds(td):
 
         attempt = 0
 
-        if not os.path.isdir(self.directory):
-            os.makedirs(self.directory)
+        directory_part = self.destination
+
+        # if file name is present in destination
+        if(self.target == self.destination):
+            directory_part = self.destination.rpartition('/')[0]
+
+        if directory_part and not os.path.isdir(directory_part):
```

Comment:
Will there be a condition when we do not have a directory? I think we can omit this check.